app/models/concerns/category_hashtag.rb

```diff
@@ -40,7 +40,7 @@ def query_loaded_from_slugs(category_slugs, categories)
               end
             end
           else
-            categories.find { |cat| cat.slug.casecmp?(parent_slug) && cat.top_level? }
+            categories.find { |cat| cat.parent_category_id.nil? && cat.slug.casecmp?(parent_slug) }
```

Comment:
Will you be able to expand on why this is being changed in the context of this PR?

Reply:
This was just a small refactoring, I can move it to a different PR.